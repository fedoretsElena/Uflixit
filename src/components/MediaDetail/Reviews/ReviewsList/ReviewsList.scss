.fadeIn {
    &-appear,
    &-enter {
        opacity: 0;

        &-active {
            opacity: 1;
            transition: 0.6s opacity;
        }
    }
}
