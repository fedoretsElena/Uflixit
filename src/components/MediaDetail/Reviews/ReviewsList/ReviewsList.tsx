import React, { useState } from "react";
import { CSSTransition, TransitionGroup } from "react-transition-group";

import ReviewItem from "../ReviewsItem";
import Review from "../../../../models/Review";
import ErrorMsg from "../../../shared/ErrorMsg";
import InfinityScroll from "../../../shared/InfinityScroll";

import "./ReviewsList.scss";

interface IProps {
    reviews: Review[];
}

const ReviewsList = ({ reviews }: IProps) => {
    const MAX: number = 5;
    let [visibleReviews, incReviews] = useState(reviews.slice(0, 5));

    function showMore(): void {
        const part = reviews.slice(0, visibleReviews.length + MAX);

        visibleReviews = part;
        incReviews(part);
    }

    return (
        <section className="border-top">
            <TransitionGroup component={null}>
                {visibleReviews.map((review, index) => (
                    <CSSTransition
                        timeout={1500}
                        appear={true}
                        classNames="fadeIn"
                        key={review.id}
                    >
                        <ReviewItem
                            review={review}
                            index={index > 9 ? index / 5 : index}
                        />
                    </CSSTransition>
                ))}
            </TransitionGroup>

            <InfinityScroll
                listenUntil={visibleReviews.length < reviews.length}
                endPageHappend={showMore}
            />

            {!reviews.length && <ErrorMsg msg="There is no reviews yet." />}
        </section>
    );
};

export default ReviewsList;
