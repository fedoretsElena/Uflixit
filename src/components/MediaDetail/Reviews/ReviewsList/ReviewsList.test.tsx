import ReviewsList from "./ReviewsList";

describe("ReviewsList component", () => {
    it("should renders correctly", () => {
        const component = shallow(<ReviewsList reviews={[]} />);

        expect(component).toMatchSnapshot();
    });
});
