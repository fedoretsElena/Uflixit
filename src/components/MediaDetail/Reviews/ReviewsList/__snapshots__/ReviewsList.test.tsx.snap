// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList component should renders correctly 1`] = `
<section
  className="border-top"
>
  <TransitionGroup
    childFactory={[Function]}
    component={null}
  />
  <InfinityScroll
    endMessage={
      <p
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <b>
          Yay! You have seen it all
        </b>
      </p>
    }
    endPageHappend={[Function]}
    listenUntil={false}
  />
  <ErrorMsg
    msg="There is no reviews yet."
  />
</section>
`;
